plugins/modules/gcp_secret_manager.py

```diff
@@ -0,0 +1,429 @@
+    retunred: success
+    type: dict
+'''
+
+################################################################################
+# Imports
+################################################################################
+
+from ansible_collections.google.cloud.plugins.module_utils.gcp_utils import (
+    navigate_hash,
+    GcpSession,
+    GcpModule,
+    GcpRequest,
+    remove_nones_from_dict,
+    replace_resource_dict,
+)
+
+import json
+# for decoding and validating secrets
+import base64
+import binascii
+
+
+def get_auth(module):
+    return GcpSession(module, 'secret-manager')
+
+def self_access_link(module):
+    return "https://secretmanager.googleapis.com/v1/projects/{project}/secrets/{name}/versions/{calc_version}:access".format(**module.params)
+
+def self_get_link(module):
+    return "https://secretmanager.googleapis.com/v1/projects/{project}/secrets/{name}/versions/{calc_version}".format(**module.params)
+
+def self_update_link(module):
+    return "https://secretmanager.googleapis.com/v1/projects/{project}/secrets/{name}/versions/{calc_version:version}".format(**module.params)
+
+def self_list_link(module):
+    return "https://secretmanager.googleapis.com/v1/projects/{project}/secrets/{name}/versions?filter=state:ENABLED".format(**module.params)
+
+def self_delete_link(module):
+    return "https://secretmanager.googleapis.com/v1/projects/{project}/secrets/{name}".format(**module.params)
+
+
+def fetch_resource(module, allow_not_found=True):
+    auth = get_auth(module)
+    # set version to the latest version because
+    # we can't be sure that "latest" is always going 
+    # to be set if secret versions get disabled
+    # see https://issuetracker.google.com/issues/286489671
+    if module.params['version'] == "latest" or module.params['version'] == 'all':
+        version_list = list_secret_versions(module)
+        latest_version = None
+        if version_list is None:
+            return None
+        
+        if "versions" in version_list:
+            latest_version = sorted(version_list['versions'], key=lambda d: d['name'])[-1]['name'].split('/')[-1]
+            module.params['calc_version'] = latest_version
+        else:
+            # if this occurs, there are no available secret versions
+            # handle the corner case that we tried to delete
+            # a secret version that doesn't exist
+            if module.params['state'] == "absent":
+                return { "action": "delete_secret" }
+
+    link = self_access_link(module)
+    access_obj = return_if_object(module, auth.get(link), allow_not_found)
+    if access_obj is None:
+        return None
+    link = self_get_link(module)
+    get_obj = return_if_object(module, auth.get(link), allow_not_found)
+    if get_obj is None:
+        return None
+    return merge_dicts(get_obj, access_obj)
+
+def merge_dicts(x, y):
+    z = x.copy()
+    z.update(y)
+    return z
+
+def snake_to_camel(snake):
```

Comment:
unused function? seems like it should be a utility anyway.

Reply:
Yes, unusued indeed.  I forgot to remove this and will do so.

 As a note, this module has someone limited parameter support in creating new secrets/secret versions.  My thinking here was to get functionality out the door and see what people use.  If people need more functionality, it could be added in the future.

Reply:
Adding feature as-needed seems totally reasonable to me!

thanks for removing the dead code.